docs/bokeh/source/docs/dev_guide/setup.rst

```diff
@@ -664,6 +663,92 @@ setting up a development environment:
     and reinstall your development environment, following the steps in this guide
     from :ref:`the beginning <contributor_guide_setup_preliminaries>`.
 
+.. dropdown:: Slow network connections when cloning
+
+    If you are experiencing slow network connections or timeouts when attempting to clone our repository, consider performing a **shallow clone**.
+    This method downloads fewer commits, which speeds up the cloning process and reduces the amount of data transferred.
+
+    Using a shallow clone can be an effective workaround for contributors with limited bandwidth or those experiencing slow cloning speeds.
+    However, be aware of its limitations and know how to convert it back to a full clone if necessary.
+
+    To create a shallow clone of the repository, run:
+
+    .. tab-set::
+
+        .. tab-item:: SSH
+
+            .. code-block:: sh
+
+                git clone --depth <number-of-commits> git@github.com:bokeh/bokeh.git
+
+        .. tab-item:: HTTPS
+
+            .. code-block:: sh
+
+                git clone --depth <number-of-commits> https://github.com/bokeh/bokeh.git
+
+    Replace ``<number-of-commits>`` with the number of commits you wish to clone.
+
+    For example, to clone only the latest commit:
+
+    .. tab-set::
+
+        .. tab-item:: SSH
+
+            .. code-block:: sh
+
+                git clone --depth 1 git@github.com:bokeh/bokeh.git
+
+        .. tab-item:: HTTPS
+
+            .. code-block:: sh
+
+                git clone --depth 1 https://github.com/bokeh/bokeh.git
+
+    If you are only interested in the history of a specific branch,
+    you can combine the --single-branch option with --depth to further limit the clone to a single branch. Run:
+
+    .. tab-set::
+
+        .. tab-item:: SSH
+
+            .. code-block:: sh
+
+                git clone --depth 1 --branch <branch-name> --single-branch git@github.com:bokeh/bokeh.git
+
+        .. tab-item:: HTTPS
+
+            .. code-block:: sh
+
+                git clone --depth 1 --branch <branch-name> --single-branch https://github.com/bokeh/bokeh.git
```

Comment:
Do we need to include HTTPS instructions? If someone is doing active development then they will need to push to their fork, which generally means an SSH clone, I think?

Reply:
I think they can still push using HTTPS, although that would require entering their credentials every time.

Reply:
Right I have just never known anyone to do that really, especially with mandatory MFA enabled. I still think it could be omitted to make the answer a bit more compact and focused, but I leave it up to your discretion

Reply:
OK. I want to leave it as is though. It doesn't cause too much noise in the docs as far as I can tell.